### DISCUSSION TITLE
Option to stay in browser instead of opening GitHub links in app

### DISCUSSION THREAD

**Discussion by stutrek** [community/community#10108] [Mobile]
I am often browsing the web on purpose and wish to continue doing so. When every GitHub link opens in the app it is very disruptive.

**Comment by cglong** [comment]
Is this for an iPhone or Android phone? On Android, you can unregister the GitHub app from opening links. I don't think a similar option exists in iOS.

**Reply by stutrek** [reply]
iPhone. Since I posted this I've had GitHub links in Safari, but I don't know what the pattern is.

**Reply by aisbergde** [reply]
How can I unregister this intent on Android?
It looks like it is set in the Chrome browser.
Because my very complicated workaround is to share the link to "copy link" and then to open in another browser then Chrome.

This forcing to open in the GitHub app is just terrible. The app often is very feature limited. There should be at least an option 'open in browser'

I even needed to remove the app in the past because of issues with some other apps trying to call something on GitHub, which was impossible to do because of the GitHub app stealing the intent.

On the other hand I need the app for authentication.

But the app should me allow to deside, if I want to open github links in the app or not.

**Reply by cat-bro** [reply]
Hi @aisbergde on my android phone I was able to disable it by going to Settings -> Apps and Notifications -> github -> Advanced -> Open by default. From there it gave me the option to stop github links from opening in the app. So far so good.

**Reply by aisbergde** [reply]
I don't have these options. I have an option to "delete default settings" but there are no default settings defined for github.
Maybe because of my Android version:
MIUI Global 13.0.1
Android: 12 SP1A.210812.016

**Reply by malikshi** [reply]
MIUI have the options but it's aren't work

**Reply by jmatsushita** [reply]
It is possible on iOS now https://docs.github.com/en/get-started/using-github/github-mobile#managing-universal-links-for-github-mobile-on-ios

**Reply by Juankydestroyer** [reply]
I think you drown in a glass of water, I do not see so difficult, nor complicated any of this
But everyone complicates things in their own way.

**Reply by Shayan165** [reply]
@aisbergde  try my solution, it works for me.

**Reply by spirin** [reply]
> It is possible on iOS now https://docs.github.com/en/get-started/using-github/github-mobile#managing-universal-links-for-github-mobile-on-ios

not working with Slack

**Reply by Duonget123** [reply]
> Is this for an iPhone or Android phone? On Android, you can unregister the GitHub app from opening links. I don't think a similar option exists in iOS.

**Reply by Duonget123** [reply]
> @aisbergde  try my solution, it works for me.

**Reply by Duonget123** [reply]
> > Is this for an iPhone or Android phone? On Android, you can unregister the GitHub app from opening links. I don't think a similar option exists in iOS.
> 
>

**Comment by alexander-kastil** [comment]
> I am often browsing the web on purpose and wish to continue doing so. When every GitHub link opens in the app it is very disruptive.

Me too ... sometimes the web view offers better support to what I am doing

**Reply by alexander-kastil** [reply]
Im using iPad pro ...

**Reply by tungmeoo** [reply]
yeu cau keo

**Comment by aisbergde** [comment]
> I am often browsing the web on purpose and wish to continue doing so. When every GitHub link opens in the app it is very disruptive.

most needed feature. The current behavior is just terrible.

**Comment by vincentkoc** [comment]
I actually want the opposite, when you are in the app and open github actions link it pushes to browser then i have to go through MFA all over again

**Reply by alexander-kastil** [reply]
Basically the should be a setting on which you can set the behavior

**Comment by rajuahmed78** [comment]
> I actually want the opposite, when you are in the app and open github actions link it pushes to browser then i have to go through MFA all over vivo 1933 use GitHub only

**Comment by jmatsushita** [comment]
It is possible on iOS now with a long press which will remember your settings https://docs.github.com/en/get-started/using-github/github-mobile#managing-universal-links-for-github-mobile-on-ios

**Reply by spirin** [reply]
not working with Slack

**Reply by spirin** [reply]
this should be a normal option in the settings

**Comment by ghost** [comment]
I have found an easy way: Open every repository link in a new tab. The browser will load the website if you actually do this. If you simply click this link, the website processes the clicking event and redirects you to GitHub mobile. (This is a bit annoying, but trust me, viewing a repository in GitHub mobile is much more frustrating...)

**Comment by radiumatic** [comment]
Links in the browser open in the app, and links in the app open in the single tab browser. 

This is ***ANNOYING***.

Fix this, please.

**Reply by disgorge69** [reply]
Amen

**Comment by Shayan165** [comment]
I may have a solution.
On Android go to settings -> manage apps > press three dots at right top corner > other settings > default apps > opening links > search GitHub > change open supported links form 'open to ask every time'

My device Miui 12

For other Android users just look for default apps in that search for anything mentioned above.

**Reply by aisbergde** [reply]
hard to find in MIUI 14, because the search in settings doesn't find it. I located it here:

Einstellungen > Apps > Apps verwalten > 3-Punkte-Menu rechts oben > andere Einstellungen > Standard-Apps > Links öffnen > Liste der installierten Apps > GitHub > Unterstützte Links öffnen > Jedes mal Fragen

translated:

Settings > Apps > Manage apps > 3-dot menu top right > Other settings > Default apps > Open links > List of installed apps > GitHub > Open supported links > Ask every time

But the settings are already set to "ask every time" and I am never asked. For example if I get a notification e-mail, containing a github link, I am not asked, when I press the link.

But what I can do: I can long press on the link, then I get a context menu containing 3 topics, and there I can copy the link and explicitly open in the browser and then I stay in the browser.

![Screenshot_2024-03-07-08-38-07-426_com.google.android.gm.jpg](https://github.com/community/community/assets/9020187/fb38effe-c1a2-45fb-8ee6-b0add55d4f52)

This often helps me, when I want to open in the browser.

**Reply by Shayan165** [reply]
@aisbergde Try changing to Don't open

**Reply by Zero-Fanker** [reply]
> @aisbergde Try changing to Don't open 

Guess what, the "don't open" open option not even works. I doubt this issue also related to MIUI .

**Comment by FreesoSaiFared** [comment]
> I may have a solution.
> On Android go to settings -> manage apps > press three dots at right top corner > other settings > default apps > opening links > search GitHub > change open supported links form 'open to ask every time'
> 
> My device Miui 12
> 
> For other Android users just look for default apps in that search for anything mentioned above.

cannot find this

**Comment by spirin** [comment]
It seems like the easiest solution on iOS right now is to just delete the app.

**Reply by cdavie-artium** [reply]
That was my solution, unfortunately.

**Comment by andrew222651** [comment]
Android 13: Settings | Apps | See all | GitHub | Open by Default

**Comment by mylesgoose** [comment]
> I am often browsing the web on purpose and wish to continue doing so. When every GitHub link opens in the app it is very disruptive.

just disable the app opening links
![Screenshot_20241001_231529_Settings](https://github.com/user-attachments/assets/2fc0c17b-b83c-45c8-ac2e-6bbd8ad36820)

**Comment by chanhyokpark** [comment]
> > I am often browsing the web on purpose and wish to continue doing so. When every GitHub link opens in the app it is very disruptive.
> 
> just disable the app opening links
> ![Screenshot_20241001_231529_Settings](https://github.com/user-attachments/assets/2fc0c17b-b83c-45c8-ac2e-6bbd8ad36820)
> 

It didn't work for me. The app opens even when that setting is disabled, and it's very annoying.

**Reply by dan-lugg** [reply]
Same here, Android 14, One UI 6.1 — that toggle has no effect still, as of today.

**Reply by guodechang38** [reply]
不 No

**Comment by MasterInQuestion** [comment]
This is much browser / OS issue.
    Somewhat irrelevant with GitHub.

    The app may add special handling indeed.
    But I don't recommend using browsers don't properly handle alike.

**Comment by chanhyokpark** [comment]
Update: It seems that google.com detects app installation(I'm not sure how this is possible), and opens it directly using URI scheme. When I tried opening the app via duckduckgo, it opened in the browser as expected. I still can't find how to disable this feature in Google.

**Reply by fuzzydude42** [reply]
Same here, in the past I was able to change the setting to open links in browser and Google Chrome on Android would respect it, now it ignores the setting.

The only work around is one listed here, open link in new tab, or use another browser, Brave and Firefox both respect the setting, even when using Google for the search, so it must be a Chrome issue.

**Reply by ttlns** [reply]
Hmm couldn't find a [crbug](https://issues.chromium.org) for this though

**Comment by ttlns** [comment]
Are there any updates to expect for this?

**Comment by stalebedew-ct** [comment]
https://faq.whatsapp.com/813798063285846/?helpref=uf_share

**Comment by starikcetin** [comment]
For people on MIUI, this worked for me:
Media management apps -> GitHub app -> Open by default -> Turn off "Open supported links"

**Reply by ttlns** [reply]
> For people on MIUI, this worked for me: Media management apps -> GitHub app -> Open by default -> Turn off "Open supported links"

Not for Google-Chrome + google search
At least in my case

e.g https://www.google.com/search?q=kaliiiiiiiii+github&oq=kaliiiiiiiii+github => click on a search result opens the app

![Screenshot_2024-12-18-16-24-41-281_com android chrome](https://github.com/user-attachments/assets/7978583f-8793-4e59-8ac4-244cb3d2699f)

**Reply by starikcetin** [reply]
I just tried opening a github link in chrome from google search and it opened in chrome instead of the app. Are you sure you modified the setting through "Media management apps" category?

**Reply by ttlns** [reply]
> I just tried opening a github link in chrome from google search 

I went over Google-Chrome, but going over `com.Google.Android.googlequicksearchbox` causes the same issue

> and it opened in chrome instead of the app. Are you sure you modified the setting through "Media management apps" category?

Yes, & I'm on
- Xiaomi 10 T Pro, MIUI 14.0.1
- Google-Chrome 131.0.6778.135 (Official Build) (64-bit) 
- GH Mobile 1.189.0

![Screenshot_2024-12-19-01-17-55-872_com android settings~2](https://github.com/user-attachments/assets/681ee455-79cb-47f5-bb0f-b89d2bb54378)

**Reply by starikcetin** [reply]
Try doing this also: Apps -> Manage Apps -> Three dots at top right -> Other settings -> Default Apps -> Opening links -> Github app -> Open supported links -> Don't open

**Reply by ttlns** [reply]
nope - doesn't do the trick either:(
Maybe this is verry specific to my env - but either way, would be neat to have this option within GH mobile

**Comment by Bashirsaba** [comment]
Github desktop version still isn't available on Microsoft app stores

**Reply by Duonget123** [reply]
https://docs.github.com/en/enterprise-cloud@latest/admin/managing-iam

**Comment by Duonget123** [comment]
https://docs.github.com/en/enterprise-cloud@latest/admin/managing-iam

**Comment by mfridman** [comment]
In Chrome within the browser (e.g., laptop) go to chrome://apps/

<img width="400" alt="CleanShot 2026-01-04 at 07 57 22@2x" src="https://github.com/user-attachments/assets/c4ef9d77-4e95-4753-9ce4-2d181bba0c2f" />

Then right-click the GitHub app and either uninstall it or go to "App settings" and change "Opening supported links".

<img width="500" alt="CleanShot 2026-01-04 at 08 03 32@2x" src="https://github.com/user-attachments/assets/46ea0e73-3d31-4313-ba51-b536479ecf5b" />

Hope this helps the next person.

**Reply by seivan** [reply]
I wish I could configure it to use the browser for every GitHub link except PR and Issues. That's the only thing that actually benefits from the app.

**Reply by cle-cle-cle** [reply]
THANK YOU
